code/modules/recycling/disposal-construction.dm

```diff
@@ -186,36 +189,26 @@
 /obj/structure/disposalconstruct/attackby(obj/item/I, mob/user, params)
 	var/nicetype = get_nice_name()
 	var/ispipe = is_pipe() // Indicates if we should change the level of this pipe
+	var/turf/T = get_turf(src)
 	add_fingerprint(user)
 
-
-	var/turf/T = src.loc
-	if(T.intact)
-		to_chat(user, "You can only attach the [nicetype] if the floor plating is removed.")
-		return
-
-	if(ptype in list(PIPE_DISPOSALS_BIN, PIPE_DISPOSALS_OUTLET, PIPE_DISPOSALS_CHUTE)) // Disposal or outlet
-		var/obj/structure/disposalpipe/trunk/CP = locate() in T
-		if(!CP) // There's no trunk
-			to_chat(user, "The [nicetype] requires a trunk underneath it in order to work.")
-			return
```

Comment:
These attackby checks need to be returned, as they're were also used in the checks below from what I can tell